### ISSUE TITLE
Error for member anonymous object in non-top level class with default type parameter

### ISSUE THREAD

**Issue by sgalles** [ceylon/ceylon-compiler#2161]
This code
```ceylon
shared void run(){ // keep this method
    class Foo<Bar=String>(){ 
        shared object b {}
    }
}
```
compiled
```
error: incompatible types
    class Foo<Bar=String>(){
              ^
  required: Foo<String>.b_
  found:    Foo<Bar>.b_
  where Bar is a type-variable:
    Bar extends Object declared in class Foo
```

**Comment by sgalles** [comment]
Ha, might be actually caused by this issue https://github.com/ceylon/ceylon-spec/issues/1299

**Comment by sgalles** [comment]
OK, this was fixed by  ceylon/ceylon-spec#1299. Closed.
